chaos/cluster.go

```diff
@@ -0,0 +1,258 @@
+package chaos
+
+import (
+	"bytes"
+	"fmt"
+	"math/rand"
+	"sync"
+	"time"
+
+	"github.com/golang/glog"
+	"golang.org/x/crypto/ssh"
+
+	"github.com/coreos/ktestutil/utils"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/pkg/api/v1"
+)
+
+const (
+	// NodeRoleMasterLabel defines the master node's label.
+	NodeRoleMasterLabel = "node-role.kubernetes.io/master"
+	// NodeRoleWorkerLabel defines the worker node's label.
+	NodeRoleWorkerLabel = "node-role.kubernetes.io/node"
+
+	rebootCmd              = "sudo reboot"
+	teardownCmd            = "sudo ifconfig eth0 down && sudo systemctl stop kubelet && sudo ifconfig eth0 up"
```

Comment:
maybe we do not need ifconfig eth0 up? after the reboot, the interface will be brought up automatically? even if the interface up is needed, it should be done after the reboot not before.